internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -919,3 +952,367 @@ func TestMachineSetReconciler_updateStatusResizedCondition(t *testing.T) {
 		})
 	}
 }
+
+func TestMachineSetReconciler_syncMachines(t *testing.T) {
+	setup := func(t *testing.T, g *WithT) (*corev1.Namespace, *clusterv1.Cluster) {
+		t.Helper()
+
+		t.Log("Creating the namespace")
+		ns, err := env.CreateNamespace(ctx, "test-machine-set-reconciler-sync-machines")
+		g.Expect(err).To(BeNil())
+
+		t.Log("Creating the Cluster")
+		cluster := &clusterv1.Cluster{ObjectMeta: metav1.ObjectMeta{Namespace: ns.Name, Name: testClusterName}}
+		g.Expect(env.Create(ctx, cluster)).To(Succeed())
+
+		t.Log("Creating the Cluster Kubeconfig Secret")
+		g.Expect(env.CreateKubeconfigSecret(ctx, cluster)).To(Succeed())
+
+		return ns, cluster
+	}
+
+	teardown := func(t *testing.T, g *WithT, ns *corev1.Namespace, cluster *clusterv1.Cluster) {
+		t.Helper()
+
+		t.Log("Deleting the Cluster")
+		g.Expect(env.Delete(ctx, cluster)).To(Succeed())
+		t.Log("Deleting the namespace")
+		g.Expect(env.Delete(ctx, ns)).To(Succeed())
+	}
+
+	t.Run("Should sync machines with the in-place mutable changes", func(t *testing.T) {
+		g := NewWithT(t)
+		namespace, testCluster := setup(t, g)
+		defer teardown(t, g, namespace, testCluster)
+
+		replicas := int32(2)
+		version := "v1.25.3"
+		duration10s := &metav1.Duration{Duration: 10 * time.Second}
+		ms := &clusterv1.MachineSet{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       "abc-123-ms-uid",
+				Name:      "ms-1",
+				Namespace: namespace.Name,
+				Labels: map[string]string{
+					"label-1": "true",
+				},
+			},
+			Spec: clusterv1.MachineSetSpec{
+				ClusterName: testCluster.Name,
+				Replicas:    &replicas,
+				Selector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"machine-set-matching-label": "true",
+					},
+				},
+				Template: clusterv1.MachineTemplateSpec{
+					ObjectMeta: clusterv1.ObjectMeta{
+						Labels: map[string]string{
+							"machine-set-matching-label": "true",
+						},
+						Annotations: map[string]string{
+							"annotation-1": "true",
+							"precedence":   "MachineSet",
+						},
+					},
+					Spec: clusterv1.MachineSpec{
+						ClusterName: testCluster.Name,
+						Version:     &version,
+						Bootstrap: clusterv1.Bootstrap{
+							ConfigRef: &corev1.ObjectReference{
+								APIVersion: "bootstrap.cluster.x-k8s.io/v1beta1",
+								Kind:       "GenericBootstrapConfigTemplate",
+								Name:       "ms-template",
+							},
+						},
+						InfrastructureRef: corev1.ObjectReference{
+							APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+							Kind:       "GenericInfrastructureMachineTemplate",
+							Name:       "ms-template",
+						},
+						NodeDrainTimeout:        duration10s,
+						NodeDeletionTimeout:     duration10s,
+						NodeVolumeDetachTimeout: duration10s,
+					},
+				},
+			},
+		}
+
+		fieldV1Map := map[string]interface{}{
+			"f:metadata": map[string]interface{}{
+				"f:name": map[string]interface{}{},
+			},
+		}
+		fieldV1, err := json.Marshal(fieldV1Map)
+		g.Expect(err).To(BeNil())
```

Comment:
```suggestion
```
Just setting managed fields doesn't work. Let's drop it here and in the two Machines
(I'll have another fix below so that manager owns the right fields)